packages/circuits/crates/libs/polynomial/src/lib.nr

```diff
@@ -98,38 +98,87 @@ impl<let N: u32> Polynomial<N> {
     }
 }
 
-/// Flattens an array of polynomials into a 1D array of Field elements.
-///
-/// This helper function serializes polynomial coefficients maintaining a deterministic
-/// ordering of coefficients to ensure reproducible challenge values.
-///
-/// # Generic Parameters
-/// * `A` - The degree of each polynomial in the array
-/// * `L` - The number of polynomials in the array
-/// * `SIZE` - The total size of the target array
-///
-/// # Arguments
-/// * `inputs` - The target array to store flattened coefficients
-/// * `poly` - Array of polynomials to flatten
-/// * `offset` - Starting position in the inputs array
-///
-/// # Returns
-/// A tuple containing:
-/// * The updated inputs array with flattened coefficients
-/// * The new offset position after flattening
-pub fn flatten<let A: u32, let L: u32, let SIZE: u32>(
-    mut inputs: [Field; SIZE],
+/// Flatten `L` polynomials (each `Polynomial<A>`) into a linear Vec<Field> of packed carriers.
```

Comment:
can we improve a bit the docs here? I am having difficulties on understanding the functions